design/sequence/finish-sd.puml

```diff
@@ -0,0 +1,46 @@
+@startuml
+
+box "worker client"
+    participant ":WorkerProxy" as wc
+    participant ":Message" as m
+end box
+
+box "Server" #LightBlue
+    participant ":ServerCommandHandler" as sch
+    participant ":Database" as db
+    participant ":Notifier" as n
+    participant ":Scheduler" as s
+end box
+
+create m 
+wc -> m ++: <<create>>
+m --> wc --: msg: Message
+
+wc ->> sch ++ : finish_job(msg)
+sch -> sch ++: get_job()
+note right of sch
+    parse msg to get the finished_job
+end note
+sch --> sch --: finished_job:Job
+
+sch -> db ++ : finish_job(finished_job)
+db -> db ++: set_wc_status(finished_job.wc_id, WCStatus.AVAILABLE)
```

Comment:
There is no `set_wc_status`, because the work machine is either connected or disconnected, and whether a job can be scheduled on the work machine depends on the scheduler.